Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -2533,4 +2533,59 @@ class HTTPClientTests: XCTestCase {
         XCTAssertEqual(info.connectionNumber, 1)
         XCTAssertEqual(info.requestNumber, 1)
     }
+
+    func testBackpressure() {
+        class BackpressureResponseDelegate: HTTPClientResponseDelegate {
+            typealias Response = Void
+            var count = 0
+            var processingBodyPart = false
+            var didntWait = false
+            var lock = Lock()
+
+            init() {}
+
+            func didReceiveHead(task: HTTPClient.Task<Response>, _ head: HTTPResponseHead) -> EventLoopFuture<Void> {
+                return task.eventLoop.makeSucceededFuture(())
+            }
+
+            func didReceiveBodyPart(task: HTTPClient.Task<Response>, _ part: ByteBuffer) -> EventLoopFuture<Void> {
```

Comment:
how are we making sure that `didReceiveBodyPart` is actually invoked multiple times? Because if it's only invoked once, then we don't know if download streaming works or not I think?

Reply:
We aren’t at the moment. I can add a test to see if it did. Is there a way we can force it?

Reply:
@adam-fowler From the NIO level: Yes, you could set `ChannelOptions.maxMessagesPerRead` to `1` and set the allocator to `FixedSizeRecvAllocator(1)` which means that NIO will read everything byte to byte. That'll cause AHC to call out multiple times.

The issue here is that through AHC's API, you can't easily mess with the underlying `Channel`... You could make it an internal test and force it that way.

The other (probably easier) option is:
- Download say 1 MB of data. NIO's default allocator will never send you more than 64k. Given that AHC doesn't change it, you can't get 1 MB in one big chunk.
- Add `XCTAssertGreaterOrEqual(numberOfCalls, 2)` or so.